scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_presenter.py

```diff
@@ -508,13 +429,132 @@ def clear_fit_information(self):
         self.update_fit_status_information_in_view()
         self.view.undo_fit_button.setEnabled(False)
 
-    @property
-    def start_x(self):
-        return self._start_x
+    def update_selected_workspace_guess(self):
+        if self.view.is_simul_fit():
+            self.update_fit_specifier_list()
+        if self.manual_selection_made:
+            guess_selection = self.selected_data
+            self.selected_data = guess_selection
+        else:
+            self.selected_data = self.get_workspace_selected_list()
 
-    @property
-    def end_x(self):
-        return self._end_x
+    def get_workspace_selected_list(self):
+        if self.context._frequency_context is not None:
+            freq = self.context._frequency_context.plot_type
+        else:
+            freq = 'None'
+
+        runs = 'All'
+        groups_and_pairs = self._get_selected_groups_and_pairs()
+        if self.view.is_simul_fit():
+            if self.view.simultaneous_fit_by == "Run":
+                runs = self.view.simultaneous_fit_by_specifier
+            elif self.view.simultaneous_fit_by == "Group/Pair":
+                groups_and_pairs = [self.view.simultaneous_fit_by_specifier]
+
+        guess_selection = []
+        for grppair in groups_and_pairs:
+            guess_selection += self.context.get_names_of_workspaces_to_fit(
+                runs=runs,
+                group_and_pair=grppair,
+                phasequad=False,
+                rebin=not self.view.fit_to_raw, freq=freq)
+
+        guess_selection = list(set(self._check_data_exists(guess_selection)))
+
+        if len(guess_selection) > 1:  # sort the list to preserve order
+            guess_selection.sort(key=self._workspace_list_sorter)
+
+        return guess_selection
+
+    def update_fit_specifier_list(self):
+        if self.view.simultaneous_fit_by == "Run":
+            flattened_run_list = [str(item) for sublist in self.context.data_context.current_runs for item in sublist]
```

Comment:
this is not very clear to me, due to the two for statements

Reply:
The run list in current runs returns a list of lists. E.g [[62260]]. So this needs two for loops to be able to extract the runs.

Reply:
can we make it clearer thats what you are doing? Maybe just add a comment along the line of what you just said